source/common/buffer/watermark_buffer.cc

```diff
@@ -16,6 +16,7 @@ namespace {
 // Effectively disables tracking as this should zero out all reasonable account
 // balances when shifted by this amount.
 constexpr uint32_t kEffectivelyDisableTrackingBitshift = 63;
+constexpr uint32_t kMaxNumberOfStreamsToResetPerInvocation = 50;
```

Comment:
I'd comment here as well that 50 is arbitrary, to limit work done (as you doc up above)

Reply:
Done.